cloud/src/autogluon/cloud/predictor/predictor.py

```diff
@@ -897,3 +951,125 @@ def _get_local_predictor_cls(self):
+
+        Parameters
+        ----------
+        test_data: Union(str, pandas.DataFrame)
+            The test data to be inferenced.
+            Can be a pandas.DataFrame, a numpy.ndarray, a local path or a s3 path to a csv file containing paths of test images.
+            Or a local path to a single image file.
+            Or a list of local paths to image files.
+        test_data_image_column: Optional(str)
+            If provided a pandas.DataFrame as the test_data, you must specify the column name corresponding to image paths.
+            Images has to live in the same directory specified by the column.
+        accept: str, default = application/x-parquet
+            Type of accept output content.
+            Valid options are application/x-parquet, text/csv, application/json
+
+        Returns
+        -------
+        Pandas.DataFrame
+        Predict results in DataFrame
+        """
+        assert self.endpoint, 'Please call `deploy()` to deploy an endpoint first.'
+        assert accept in VALID_ACCEPT, f'Invalid accept type. Options are {VALID_ACCEPT}.'
+
+        import cv2
+        import numpy as np
+
+        if isinstance(test_data, str):
+            image = cv2.imread(test_data)
+            if image is None:  # not an image
+                test_data = load_pd.load(test_data)
+            else:
+                test_data = image
+        if isinstance(test_data, list):
+            images = []
+            for image in test_data:
+                images.append(cv2.imread(image))
+            test_data = np.array(images, dtype=object)
+        if isinstance(test_data, pd.DataFrame):
+            assert test_data_image_column is not None, 'Please specify an image column name'
+            assert test_data_image_column in test_data, 'Please specify a valid image column name'
+
+            # Convert test data to be numpy array for network transfer
+            test_data = test_data[test_data_image_column].apply(lambda path: cv2.imread(path)).to_numpy()
+        assert isinstance(test_data, np.ndarray), f'Invalid test data format {type(test_data)}'
+
+        return self._predict_realtime(test_data=test_data, accept=accept)
+
+    def _upload_batch_predict_data(self, test_data, bucket, key_prefix):
+        if not is_s3_url(test_data):
+            if not os.path.isdir(test_data):
+                logger.warning('Are you sure you want to do batch inference on a single image? You might want to try `deploy()` and `predict_realtime()` instead')
+            logger.log(20, 'Uploading data...')
+            test_input = self.sagemaker_session.upload_data(
+                path=test_data,
+                bucket=bucket,
+                key_prefix=key_prefix + '/data'
+            )
+            logger.log(20, 'Data uploaded successfully')
+        else:
+            test_input = test_data
+        return test_input
+
+    def predict(
+        self,
+        test_data,
+        transformer_kwargs=dict(),
```

Comment:
Do not use dict() as default, it is mutable!! (aka your default can be mutated and not be empty anymore in future calls...) Instead default=None and then set to dict() inside of method.